.github/workflows/create-release.yaml

```diff
@@ -1,53 +1,280 @@
 name: "Create release"
-run-name: "Create release ${{ inputs.release_version }}"
-
-permissions:
-  contents: write
+run-name: "Create release ${{ inputs.name }} force: ${{ inputs.force }} dry-run: ${{ inputs.dry-run }}"
 
 on:
   workflow_dispatch:
     inputs:
-      release_version:
-        description: "Release version (i.e. x.y.z)"
+      name: # release tag
+        description: "Release tag (i.e. x.y.z)"
         default: ""
         required: true
+      force:
+        description: "Force the release creation. This will overwrite an existing release with the same tag."
+        default: false
+        type: boolean
       dry-run:
         type: boolean
-        description: "Creates a dry-run branch and tag. This will not trigger the actual release process."
-        default: true
+        description: "Do not publish. Builds image and runs tests but does not publish the release. (dry-run)"
+        default: false
+#      create-pr:
+#        type: boolean
+#        description: "Create PR to management-plane-charts"
+#        default: true
+
+permissions:
+  id-token: write # This is required for requesting the JWT token
+  contents: write # This is required for creating a release
+
+env:
+  GH_HOST: github.com
+  GH_TOKEN: ${{ github.token }}
+  GH_REPO: ${{ github.repository }}
+  IMAGE_REGISTRY: europe-docker.pkg.dev/kyma-project/prod
+  IMAGE_NAME: kyma-otel-collector
 
 jobs:
-  # check if tag is a valid semantic version
-  # check that no release already exists with that name
   validate-release:
     name: Validate release
     runs-on: ubuntu-latest
     steps:
-      # checkout code including all history
       - name: Checkout code
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
         with:
           fetch-depth: 0
-          ref: main
+          ref: ${{ github.ref }}
 
       - name: Check if that release tag is correctly formatted
-        run: "./hack/scripts/check_release_tag.sh ${{ inputs.release_version }}"
+        run: |
+          # Validate that the release tag is a proper semantic version
+          if ! [[ ${{ inputs.name }} =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "::error ::Release tag ${{ inputs.name }} is not a valid semantic version"
+            exit 1
+          fi
+
+      - name: Do NOT allow overriding a release when it is not a draft
```

Comment:
In line 12, you use the term "overwrite"; here, it is "override". Sometimes people mix those up. I trust that you are using different words here because you mean different processes.